lib/WasmReader/WasmBinaryReader.cpp

```diff
@@ -167,6 +178,35 @@ WasmBinaryReader::ReadFromModule()
      return wnLIMIT;
 }
 
+SectionCode
+WasmBinaryReader::SectionHeader()
+{
+    UINT len;
+    UINT sectionSize;
+    UINT idSize;
+
+    sectionSize = LEB128(len);
+    m_moduleState.count += len;
+    idSize = LEB128(len);
+    m_moduleState.count += len;
+    const char *sectionName = (char*)(m_pc);
+    m_pc += idSize;
+    m_moduleState.count += idSize;
+
+    auto cmp = [&sectionName, &idSize](const char* n) { return !strncmp(n, sectionName, idSize); };
```

Comment:
this is ok for now, but I think we will eventually want to flip around, such that you cmp "expected" section with the actual, and return invalid if not expected. also, technically these names don't need to be valid utf-8 (e.g. embedded null is legal), so we should do memcmp instead